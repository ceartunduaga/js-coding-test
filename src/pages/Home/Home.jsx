import React from "react";
import { useSelector } from 'react-redux';
import { Typography, Card, CardContent, Grid } from '@mui/material';
import { BarChartCustome } from "../../components/BarChartCustome/BarChartCustome";
import { PieChartCustome } from "../../components/PieChartCustome/PieChartCustome";
import { Link } from 'react-router-dom';

import styles from './HomePage.module.css';


export default function Home() {
  const { orders } = useSelector((state) => state.orders);
  const totalOrders = orders.length;
  const totalRevenue = orders.reduce((acc, order) => acc + parseFloat(order.Total.replace('$', '')), 0).toFixed(2);
  const uniqueCustomers = [...new Set(orders.map(order => order.CustomerName))];
  const totalUniqueCustomers = uniqueCustomers.length;
  const ordersPerCustomer = orders.reduce((acc, order) => {
    acc[order.CustomerName] = (acc[order.CustomerName] || 0) + 1;
    return acc;
  }, {});
  const itemQuantities = orders.flatMap(order => order.Items).reduce((acc, item) => {
    acc[item.Item] = (acc[item.Item] || 0) + parseInt(item.Quantity);
    return acc;
  }, {});
  const mostSoldItem = Object.keys(itemQuantities).length > 0
    ? Object.keys(itemQuantities).reduce((a, b) => itemQuantities[a] > itemQuantities[b] ? a : b)
    : 'No items sold';
  const revenuePerCustomer = orders.reduce((acc, order) => {
    acc[order.CustomerName] = (acc[order.CustomerName] || 0) + parseFloat(order.Total.replace('$', ''));
    return acc;
  }, {});
  const bestCustomer = Object.keys(revenuePerCustomer).length > 0
    ? Object.keys(revenuePerCustomer).reduce((a, b) => revenuePerCustomer[a] > revenuePerCustomer[b] ? a : b)
    : 'No items sold';


  const orderData = Object.entries(ordersPerCustomer).map(([customer, orders]) => ({
    customer,
    orders,
    revenue: revenuePerCustomer[customer]
  }));

  return (
    <div className={styles.container}>
      <Typography variant="h4" sx={{ mb: 2 }}>
        Order Statistics
      </Typography>
      <Grid container spacing={2}>
        <Grid item xs={12} md={4}>
          <Card variant="outlined">
            <CardContent>
              <Typography variant="h3" color="primary">
                {totalOrders}
              </Typography>
              <Typography variant="h6">Total Orders</Typography>
            </CardContent>
          </Card>
        </Grid>
        <Grid item xs={12} md={4}>
          <Card variant="outlined">
            <CardContent>
              <Typography variant="h3" color="primary">
                $ {totalRevenue}
              </Typography>
              <Typography variant="h6">Total Revenue</Typography>
            </CardContent>
          </Card>
        </Grid>
        <Grid item xs={12} md={4}>
          <Card variant="outlined">
            <CardContent>
              <Typography variant="h3" color="primary">
                {totalUniqueCustomers}
              </Typography>
              <Typography variant="h6">Unique Customers</Typography>
            </CardContent>
          </Card>
        </Grid>
        <Grid item xs={12} md={4}>
          <Card variant="outlined">
            <CardContent>
              <Typography variant="h3" color="primary"
                component={Link}
                to={`/item/${mostSoldItem}`}>
                {mostSoldItem}
              </Typography>
              <Typography variant="h6">Most Sold Item</Typography>
            </CardContent>
          </Card>
        </Grid>
        <Grid item xs={12} md={4}>
          <Card variant="outlined">
            <CardContent>
              <Typography variant="h3" color="primary"
                component={Link}
                to={`/customer/${bestCustomer}`}
                >
                {bestCustomer}
              </Typography>
              <Typography variant="h6">Best Customer</Typography>
            </CardContent>
          </Card>
        </Grid>
        <Grid item xs={12} md={4}>
          <Card variant="outlined">
            <CardContent>
              <Typography variant="h3" color="primary">
                $ {revenuePerCustomer[bestCustomer]?.toFixed(2)}
              </Typography>
              <Typography variant="h6">Revenue Generated by Best Customer</Typography>
            </CardContent>
          </Card>
        </Grid>
        <Grid item xs={12} md={8}>
          <Typography variant="h4" mt={4}>Detailed Statistics</Typography>
          <BarChartCustome  order={orderData}/>
        </Grid>
        <Grid item xs={12} md={4}>
          <Typography variant="h4" mt={4}>Detail of items by units ordered</Typography>
          <PieChartCustome items={itemQuantities} />
        </Grid>
      </Grid>
    </div>
  );
}
