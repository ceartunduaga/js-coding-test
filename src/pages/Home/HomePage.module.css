.title {
color: #333;
}

.list {
list-style-type: none;
padding: 0;
}

.list-item {
padding: 10px;
border-bottom: 1px solid #ccc;
}