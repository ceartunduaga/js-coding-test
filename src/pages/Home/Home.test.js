import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import Home from './Home';

const mockStore = configureStore([]);
const initialState = {
  orders: {
    orders: [
      {
        OrderId: 1,
        CustomerId: 1,
        CustomerName: "Elizabeth",
        Total: "$30.00",
        Date: "2021-02-01 08:30:00.000",
        Items: [
          { Item: "Candle", ItemPrice: "$3.00", Quantity: "3" },
          { Item: "Book", ItemPrice: "$15.00", Quantity: "1" },
          { Item: "Pen", ItemPrice: "$0.75", Quantity: "1" },
          { Item: "Paper", ItemPrice: "$5.25", Quantity: "1" },
        ],
      },
      {
        OrderId: 2,
        CustomerId: 2,
        CustomerName: "Alexander",
        Total: "$52.50",
        Date: "2021-02-02 10:00:00.000",
        Items: [
          { Item: "Book", ItemPrice: "$15.00", Quantity: "1" },
          { Item: "Jar", ItemPrice: "$12.50", Quantity: "3" },
        ],
      },
    ],
    loading: false,
    error: null,
  },
};

describe('Home component', () => {
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('should match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Home />
        </MemoryRouter>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
