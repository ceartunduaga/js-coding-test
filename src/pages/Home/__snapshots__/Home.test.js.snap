// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 css-pxw7rc-MuiTypography-root"
    >
      Order Statistics
    </h4>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3 css-1biy274-MuiTypography-root"
            >
              2
            </h3>
            <h6
              class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            >
              Total Orders
            </h6>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3 css-1biy274-MuiTypography-root"
            >
              $ 82.50
            </h3>
            <h6
              class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            >
              Total Revenue
            </h6>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3 css-1biy274-MuiTypography-root"
            >
              2
            </h3>
            <h6
              class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            >
              Unique Customers
            </h6>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <a
              class="MuiTypography-root MuiTypography-h3 css-1biy274-MuiTypography-root"
              href="/item/Jar"
            >
              Jar
            </a>
            <h6
              class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            >
              Most Sold Item
            </h6>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <a
              class="MuiTypography-root MuiTypography-h3 css-1biy274-MuiTypography-root"
              href="/customer/Alexander"
            >
              Alexander
            </a>
            <h6
              class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            >
              Best Customer
            </h6>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3 css-1biy274-MuiTypography-root"
            >
              $ 52.50
            </h3>
            <h6
              class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            >
              Revenue Generated by Best Customer
            </h6>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 css-9drkqd-MuiGrid-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-4fy7yj-MuiTypography-root"
        >
          Detailed Statistics
        </h4>
        <div
          class="css-1gakkqb-MuiResponsiveChart-container"
        >
          <svg
            class="css-13aj3tc-MuiChartsSurface-root"
            height="400"
            viewBox="0 0 100 400"
            width="100"
          >
            <title />
            <desc />
            <defs />
            <g
              clip-path="url(#:r0:-clip-path)"
            >
              <rect
                class="MuiBarElement-root MuiBarElement-series-auto-generated-id-0 css-sp7k3i-MuiBarElement-root"
                cursor="unset"
                style="height: 0px; width: 0px; transform: translate3d(50px,350px,0);"
              />
              <rect
                class="MuiBarElement-root MuiBarElement-series-auto-generated-id-0 css-sp7k3i-MuiBarElement-root"
                cursor="unset"
                style="height: 0px; width: 0px; transform: translate3d(50px,350px,0);"
              />
              <rect
                class="MuiBarElement-root MuiBarElement-series-auto-generated-id-1 css-qlregk-MuiBarElement-root"
                cursor="unset"
                style="height: 0px; width: 0px; transform: translate3d(50px,350px,0);"
              />
              <rect
                class="MuiBarElement-root MuiBarElement-series-auto-generated-id-1 css-qlregk-MuiBarElement-root"
                cursor="unset"
                style="height: 0px; width: 0px; transform: translate3d(50px,350px,0);"
              />
            </g>
            <g
              class="MuiChartsAxis-root MuiChartsAxis-directionX MuiChartsAxis-bottom css-1k2u9zb-MuiChartsAxis-root"
              transform="translate(0, 350)"
            >
              <line
                class="MuiChartsAxis-line"
                x1="50"
                x2="50"
              />
              <g
                class="MuiChartsAxis-tickContainer"
                transform="translate(50, 0)"
              >
                <line
                  class="MuiChartsAxis-tick"
                  y2="6"
                />
                <text
                  class="MuiChartsAxis-tickLabel"
                  dominant-baseline="hanging"
                  style="font-size: 12px;"
                  text-anchor="middle"
                  x="0"
                  y="9"
                >
                  <tspan
                    dominant-baseline="hanging"
                    dy="0px"
                    x="0"
                  >
                    Elizabeth
                  </tspan>
                </text>
              </g>
              <g
                class="MuiChartsAxis-tickContainer"
                transform="translate(50, 0)"
              >
                <line
                  class="MuiChartsAxis-tick"
                  y2="6"
                />
                <text
                  class="MuiChartsAxis-tickLabel"
                  dominant-baseline="hanging"
                  style="font-size: 12px;"
                  text-anchor="middle"
                  x="0"
                  y="9"
                >
                  <tspan
                    dominant-baseline="hanging"
                    dy="0px"
                    x="0"
                  >
                    Alexander
                  </tspan>
                </text>
              </g>
            </g>
            <g
              class="MuiChartsAxis-root MuiChartsAxis-directionY MuiChartsAxis-left css-1k2u9zb-MuiChartsAxis-root"
              transform="translate(50, 0)"
            >
              <line
                class="MuiChartsAxis-line"
                y1="350"
                y2="50"
              />
              <g
                class="MuiChartsAxis-tickContainer"
                transform="translate(0, 350)"
              >
                <line
                  class="MuiChartsAxis-tick"
                  x2="-6"
                />
                <text
                  class="MuiChartsAxis-tickLabel"
                  dominant-baseline="central"
                  style="font-size: 12px;"
                  text-anchor="end"
                  x="-8"
                  y="0"
                >
                  <tspan
                    dominant-baseline="central"
                    dy="0px"
                    x="-8"
                  >
                    0
                  </tspan>
                </text>
              </g>
              <g
                class="MuiChartsAxis-tickContainer"
                transform="translate(0, 300)"
              >
                <line
                  class="MuiChartsAxis-tick"
                  x2="-6"
                />
                <text
                  class="MuiChartsAxis-tickLabel"
                  dominant-baseline="central"
                  style="font-size: 12px;"
                  text-anchor="end"
                  x="-8"
                  y="0"
                >
                  <tspan
                    dominant-baseline="central"
                    dy="0px"
                    x="-8"
                  >
                    10
                  </tspan>
                </text>
              </g>
              <g
                class="MuiChartsAxis-tickContainer"
                transform="translate(0, 250.00000000000003)"
              >
                <line
                  class="MuiChartsAxis-tick"
                  x2="-6"
                />
                <text
                  class="MuiChartsAxis-tickLabel"
                  dominant-baseline="central"
                  style="font-size: 12px;"
                  text-anchor="end"
                  x="-8"
                  y="0"
                >
                  <tspan
                    dominant-baseline="central"
                    dy="0px"
                    x="-8"
                  >
                    20
                  </tspan>
                </text>
              </g>
              <g
                class="MuiChartsAxis-tickContainer"
                transform="translate(0, 200)"
              >
                <line
                  class="MuiChartsAxis-tick"
                  x2="-6"
                />
                <text
                  class="MuiChartsAxis-tickLabel"
                  dominant-baseline="central"
                  style="font-size: 12px;"
                  text-anchor="end"
                  x="-8"
                  y="0"
                >
                  <tspan
                    dominant-baseline="central"
                    dy="0px"
                    x="-8"
                  >
                    30
                  </tspan>
                </text>
              </g>
              <g
                class="MuiChartsAxis-tickContainer"
                transform="translate(0, 150)"
              >
                <line
                  class="MuiChartsAxis-tick"
                  x2="-6"
                />
                <text
                  class="MuiChartsAxis-tickLabel"
                  dominant-baseline="central"
                  style="font-size: 12px;"
                  text-anchor="end"
                  x="-8"
                  y="0"
                >
                  <tspan
                    dominant-baseline="central"
                    dy="0px"
                    x="-8"
                  >
                    40
                  </tspan>
                </text>
              </g>
              <g
                class="MuiChartsAxis-tickContainer"
                transform="translate(0, 100)"
              >
                <line
                  class="MuiChartsAxis-tick"
                  x2="-6"
                />
                <text
                  class="MuiChartsAxis-tickLabel"
                  dominant-baseline="central"
                  style="font-size: 12px;"
                  text-anchor="end"
                  x="-8"
                  y="0"
                >
                  <tspan
                    dominant-baseline="central"
                    dy="0px"
                    x="-8"
                  >
                    50
                  </tspan>
                </text>
              </g>
              <g
                class="MuiChartsAxis-tickContainer"
                transform="translate(0, 50)"
              >
                <line
                  class="MuiChartsAxis-tick"
                  x2="-6"
                />
                <text
                  class="MuiChartsAxis-tickLabel"
                  dominant-baseline="central"
                  style="font-size: 12px;"
                  text-anchor="end"
                  x="-8"
                  y="0"
                >
                  <tspan
                    dominant-baseline="central"
                    dy="0px"
                    x="-8"
                  >
                    60
                  </tspan>
                </text>
              </g>
            </g>
            <g
              class="MuiChartsLegend-root MuiChartsLegend-row css-1u0lry5-MuiChartsLegend-root"
            >
              <g
                class="MuiChartsLegend-series"
                transform="translate(15 25)"
              >
                <rect
                  class="MuiChartsLegend-mark"
                  fill="rgba(75, 192, 192, 0.6)"
                  height="20"
                  width="20"
                  x="0"
                  y="-10"
                />
                <text
                  dominant-baseline="central"
                  style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
                  text-anchor="start"
                  x="25"
                  y="0"
                >
                  <tspan
                    dominant-baseline="central"
                    dy="0px"
                    x="25"
                  >
                    Total Orders
                  </tspan>
                </text>
              </g>
              <g
                class="MuiChartsLegend-series"
                transform="translate(50 25)"
              >
                <rect
                  class="MuiChartsLegend-mark"
                  fill="rgba(54, 162, 235, 0.6)"
                  height="20"
                  width="20"
                  x="0"
                  y="-10"
                />
                <text
                  dominant-baseline="central"
                  style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
                  text-anchor="start"
                  x="25"
                  y="0"
                >
                  <tspan
                    dominant-baseline="central"
                    dy="0px"
                    x="25"
                  >
                    Total Revenues
                  </tspan>
                </text>
              </g>
            </g>
            <clippath
              id=":r0:-clip-path"
            >
              <rect
                height="300"
                width="0"
                x="50"
                y="50"
              />
            </clippath>
          </svg>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-4fy7yj-MuiTypography-root"
        >
          Detail of items by units ordered
        </h4>
        <div
          class="css-1n4epfq-MuiResponsiveChart-container"
        >
          <svg
            class="css-13aj3tc-MuiChartsSurface-root"
            height="350"
            viewBox="0 0 400 350"
            width="400"
          >
            <title />
            <desc />
            <defs />
            <g>
              <g
                transform="translate(152.5, 175)"
              >
                <g>
                  <path
                    class="MuiPieArc-root MuiPieArc-series-auto-generated-id-0 css-1mhcdve-MuiPieArc-root"
                    cursor="unset"
                    d="M59.665,-43.349L0,0Z"
                    style="fill: #02B2AF; opacity: 1;"
                  />
                  <path
                    class="MuiPieArc-root MuiPieArc-series-auto-generated-id-0 css-1mhcdve-MuiPieArc-root"
                    cursor="unset"
                    d="M43.349,59.665L0,0Z"
                    style="fill: #2E96FF; opacity: 1;"
                  />
                  <path
                    class="MuiPieArc-root MuiPieArc-series-auto-generated-id-0 css-1mhcdve-MuiPieArc-root"
                    cursor="unset"
                    d="M-22.79,70.14L0,0Z"
                    style="fill: #B800D8; opacity: 1;"
                  />
                  <path
                    class="MuiPieArc-root MuiPieArc-series-auto-generated-id-0 css-1mhcdve-MuiPieArc-root"
                    cursor="unset"
                    d="M-59.665,43.349L0,0Z"
                    style="fill: #60009B; opacity: 1;"
                  />
                  <path
                    class="MuiPieArc-root MuiPieArc-series-auto-generated-id-0 css-1mhcdve-MuiPieArc-root"
                    cursor="unset"
                    d="M-59.665,-43.349L0,0Z"
                    style="fill: #2731C8; opacity: 1;"
                  />
                </g>
              </g>
              <g
                transform="translate(152.5, 175)"
              >
                <g>
                  <text
                    class="MuiPieArcLabel-root MuiPieArcLabel-series-auto-generated-id-0 css-ixpz1i-MuiPieArcLabel-root"
                    style="opacity: 0; transform: none;"
                  />
                  <text
                    class="MuiPieArcLabel-root MuiPieArcLabel-series-auto-generated-id-0 css-ixpz1i-MuiPieArcLabel-root"
                    style="opacity: 0; transform: none;"
                  />
                  <text
                    class="MuiPieArcLabel-root MuiPieArcLabel-series-auto-generated-id-0 css-ixpz1i-MuiPieArcLabel-root"
                    style="opacity: 0; transform: none;"
                  />
                  <text
                    class="MuiPieArcLabel-root MuiPieArcLabel-series-auto-generated-id-0 css-ixpz1i-MuiPieArcLabel-root"
                    style="opacity: 0; transform: none;"
                  />
                  <text
                    class="MuiPieArcLabel-root MuiPieArcLabel-series-auto-generated-id-0 css-ixpz1i-MuiPieArcLabel-root"
                    style="opacity: 0; transform: none;"
                  />
                </g>
              </g>
            </g>
            <g
              class="MuiChartsLegend-root MuiChartsLegend-column css-1u0lry5-MuiChartsLegend-root"
            >
              <g
                class="MuiChartsLegend-series"
                transform="translate(355 115)"
              >
                <rect
                  class="MuiChartsLegend-mark"
                  fill="#02B2AF"
                  height="20"
                  width="20"
                  x="0"
                  y="-10"
                />
                <text
                  dominant-baseline="central"
                  style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
                  text-anchor="start"
                  x="25"
                  y="0"
                >
                  <tspan
                    dominant-baseline="central"
                    dy="0px"
                    x="25"
                  >
                    Candle
                  </tspan>
                </text>
              </g>
              <g
                class="MuiChartsLegend-series"
                transform="translate(355 145)"
              >
                <rect
                  class="MuiChartsLegend-mark"
                  fill="#2E96FF"
                  height="20"
                  width="20"
                  x="0"
                  y="-10"
                />
                <text
                  dominant-baseline="central"
                  style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
                  text-anchor="start"
                  x="25"
                  y="0"
                >
                  <tspan
                    dominant-baseline="central"
                    dy="0px"
                    x="25"
                  >
                    Book
                  </tspan>
                </text>
              </g>
              <g
                class="MuiChartsLegend-series"
                transform="translate(355 175)"
              >
                <rect
                  class="MuiChartsLegend-mark"
                  fill="#B800D8"
                  height="20"
                  width="20"
                  x="0"
                  y="-10"
                />
                <text
                  dominant-baseline="central"
                  style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
                  text-anchor="start"
                  x="25"
                  y="0"
                >
                  <tspan
                    dominant-baseline="central"
                    dy="0px"
                    x="25"
                  >
                    Pen
                  </tspan>
                </text>
              </g>
              <g
                class="MuiChartsLegend-series"
                transform="translate(355 205)"
              >
                <rect
                  class="MuiChartsLegend-mark"
                  fill="#60009B"
                  height="20"
                  width="20"
                  x="0"
                  y="-10"
                />
                <text
                  dominant-baseline="central"
                  style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
                  text-anchor="start"
                  x="25"
                  y="0"
                >
                  <tspan
                    dominant-baseline="central"
                    dy="0px"
                    x="25"
                  >
                    Paper
                  </tspan>
                </text>
              </g>
              <g
                class="MuiChartsLegend-series"
                transform="translate(355 235)"
              >
                <rect
                  class="MuiChartsLegend-mark"
                  fill="#2731C8"
                  height="20"
                  width="20"
                  x="0"
                  y="-10"
                />
                <text
                  dominant-baseline="central"
                  style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
                  text-anchor="start"
                  x="25"
                  y="0"
                >
                  <tspan
                    dominant-baseline="central"
                    dy="0px"
                    x="25"
                  >
                    Jar
                  </tspan>
                </text>
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
